tcms/core/middleware.py

```diff
@@ -28,7 +28,7 @@ def process_request(self, request):
 class CheckSettingsMiddleware(MiddlewareMixin):
     def process_request(self, request):
         doc_url = "https://kiwitcms.readthedocs.io/en/latest/admin.html#configure-kiwi-s-base-url"
-        if request.path == "/init-db/":
+        if request.path != "/":
             return
```

Comment:
This doesn't speed things up a single bit b/c the middleware is still active and still checking every request before it decides what to do. Remove the middleware classes and move all the checks inside the dashboard view.